Improve test error messages when product not found

When people write a new checker for the first time, they often are confused when their first attempt at longtests gives them an error message like this:

```python
AssertionError: assert 'newchecker' in {'glibc'}
```

This happens here https://github.com/intel/cve-bin-tool/blob/38d497a3bc00df3a45d516f7f7de45ef238ca862/test/test_scanner.py#L254
```python
        assert product in list_products
```

(And similar in a few other places)

Pytest has some ability to do this (see the docs https://docs.pytest.org/en/stable/assert.html) so I'm going to add an error message. 

(No one else needs to work on this issue; I've already got a PR ready to go)

--- a/test/test_scanner.py
+++ b/test/test_scanner.py
@@ -251,7 +251,10 @@ class TestScanner:
                 list_versions.add(product_info.version)
 
         # Make sure the product and version are in the results
-        assert product in list_products
+        assert (
+            product in list_products
+        ), f"""{product} not found in {package_name}.
+        The checker signature or url may be incorrect."""
         assert version in list_versions
 
     def test_does_not_scan_symlinks(self):

